programs/uxd/src/instructions/credix_lp/rebalance_request_execute_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,646 @@
+    );
+
+    // ---------------------------------------------------------------------
+    // -- Phase 2
+    // -- Fetch all current onchain state
+    // ---------------------------------------------------------------------
+
+    let credix_lp_depository_collateral_amount_before: u64 =
+        ctx.accounts.depository_collateral.amount;
+    let identity_depository_collateral_amount_before: u64 =
+        ctx.accounts.identity_depository_collateral.amount;
+    let profits_beneficiary_collateral_amount_before: u64 =
+        ctx.accounts.profits_beneficiary_collateral.amount;
+
+    let liquidity_collateral_amount_before: u64 = ctx.accounts.credix_liquidity_collateral.amount;
+    let outstanding_collateral_amount_before: u64 = ctx
+        .accounts
+        .credix_global_market_state
+        .pool_outstanding_credit;
+
+    let total_shares_supply_before: u64 = ctx.accounts.credix_shares_mint.supply;
+    let total_shares_value_before: u64 = liquidity_collateral_amount_before
+        .checked_add(outstanding_collateral_amount_before)
+        .ok_or(UxdError::MathError)?;
+
+    let owned_shares_amount_before: u64 = ctx.accounts.depository_shares.amount;
+    let owned_shares_value_before: u64 = compute_value_for_shares_amount_floor(
+        owned_shares_amount_before,
+        total_shares_supply_before,
+        total_shares_value_before,
+    )?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 3
+    // -- Compute the profits and the overflow of the depository we will want to withdraw
+    // ---------------------------------------------------------------------
+
+    let redeemable_amount_under_management = checked_convert_u128_to_u64(
+        ctx.accounts
+            .depository
+            .load()?
+            .redeemable_amount_under_management,
+    )?;
+
+    let profits_collateral_amount = owned_shares_value_before
+        .checked_sub(redeemable_amount_under_management)
+        .ok_or(UxdError::MathError)?;
+
+    let overflow_value = {
+        let redeemable_amount_under_management_target_amount =
+            calculate_credix_lp_depository_target_amount(
+                &ctx.accounts.controller,
+                &ctx.accounts.identity_depository,
+                &ctx.accounts.mercurial_vault_depository,
+                &ctx.accounts.depository,
+            )?;
+        if redeemable_amount_under_management < redeemable_amount_under_management_target_amount {
+            0
+        } else {
+            redeemable_amount_under_management
+                .checked_sub(redeemable_amount_under_management_target_amount)
+                .ok_or(UxdError::MathError)?
+        }
+    };
+
+    // ---------------------------------------------------------------------
+    // -- Phase 4
+    // -- Here we compute how much we would like to withdraw now,
+    // -- We also need to compute how much credix will let us withdraw now,
```

Comment:
What if there is not enough when you call this ix? Can you call it again if credix adds more collateral to the pot few hours later and get the rest?

Does this actual ix handle that scenario?

Reply:
Yeah the maths handle this scenario (integration test reflects that in the _illiquid test variant).

Basically the pot is deterministic, based on the maths for the total_withdrawal_amount we know exactly how much we can withdraw now.

And the maths always do compute `min(available, desired)` of all the amounts to account for this